index.js

```diff
@@ -6,22 +6,39 @@ const chalk = require('chalk');
 const prettyBytes = require('pretty-bytes');
 const StreamCounter = require('stream-counter');
 const gzipSize = require('gzip-size');
+const brotliSize = require('brotli-size');
 
-module.exports = options => {
+module.exports = (options = {}) => {
 	options = {
 		pretty: true,
 		showTotal: true,
+		uncompressed: options.uncompressed || !(options.gzip || options.brotli),
 		...options
 	};
 
-	let totalSize = 0;
 	let fileCount = 0;
+	const totalSize = new Map();
 
-	function log(what, size) {
+	function log(what, sizes) {
 		let {title} = options;
 		title = title ? chalk.cyan(title) + ' ' : '';
-		size = options.pretty ? prettyBytes(size) : (size + ' B');
-		fancyLog(title + what + ' ' + chalk.magenta(size) + (options.gzip ? chalk.gray(' (gzipped)') : ''));
+		const sizeStrings = [...sizes].map(([key, size]) => {
+			size = options.pretty ? prettyBytes(size) : (size + ' B');
+			return chalk.magenta(size) + chalk.gray(key);
+		});
+
+		fancyLog(title + what + ' ' + sizeStrings.join(chalk.magenta(', ')));
+	}
+
+	function hasSize(sizes) {
+		return [...sizes.values()].some(size => size > 0);
+	}
+
+	function promisify(stream, property, event = 'end') {
+		return new Promise((resolve, reject) => {
+			stream.on(event, () => resolve(stream[property]));
+			stream.on('error', error => reject(error));
+		});
 	}
```

Comment:
These functions can be at the top-level. They don't actually need any state from inside the function.